OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -34,6 +34,10 @@ public class BuildingInfo : ITraitInfo, IOccupySpaceInfo, IPlaceBuildingDecorati
 		public readonly CVec Dimensions = new CVec(1, 1);
 		public readonly bool RequiresBaseProvider = false;
 		public readonly bool AllowInvalidPlacement = false;
+		[Desc("clear smudges before add building")]
+		public readonly bool RemoveSmudgesOnBuild = true;
+		[Desc("clear smudges when selling")]
+		public readonly bool RemoveSmudgesOnSell = true;
```

Comment:
OnSell added becase of #4791. Not sure is it very useful, but adding that flag is easy so why not